src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -3788,5 +3787,39 @@ class Test
             var symbolInfo = specModel.GetSymbolInfo(newAttributeUsage.ArgumentList.Arguments[0].Expression);
             Assert.Equal(SpecialType.System_String, ((IFieldSymbol)symbolInfo.Symbol).Type.SpecialType);
         }
+
+        [Fact]
+        [WorkItem(38638, "https://github.com/dotnet/roslyn/issues/38638")]
+        public void TypeParameter_Default()
+        {
+            var source =
+@"#nullable enable
+abstract class A<T>
+{
+    internal abstract void F<U>() where U : T;
+}
+class B1<T> : A<T>
+{
+    internal override void F<U>() { _ = default(U); }
+}
+class B2 : A<int?>
+{
+    internal override void F<U>() { _ = default(U); }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var exprs = tree.GetRoot().DescendantNodes().OfType<DefaultExpressionSyntax>().ToArray();
+            verify(exprs[0], PublicNullableAnnotation.NotAnnotated, PublicNullableFlowState.MaybeNull);
+            verify(exprs[1], PublicNullableAnnotation.Annotated, PublicNullableFlowState.MaybeNull);
+
+            void verify(DefaultExpressionSyntax expr, PublicNullableAnnotation expectedAnnotation, PublicNullableFlowState expectedState)
```

Comment:
consider putting expected values first or labeling in order to match the conventions of the xunit Assert methods.

Reply:
I'll leave as is since this matches similar local functions in this class.

---
In reply to: [348226625](https://github.com/dotnet/roslyn/pull/39778#discussion_r348226625) [](ancestors = 348226625)